src/platform/packages/private/kbn-grid-layout/grid/grid_layout.tsx

```diff
@@ -84,7 +83,7 @@ export const GridLayout = ({
      * The only thing that should cause the entire layout to re-render is adding a new row;
      * this subscription ensures this by updating the `rowCount` state when it changes.
      */
-    const rowCountSubscription = gridLayoutStateManager.gridLayout$
+    const rowCountSubscription = gridLayoutStateManager.proposedGridLayout$
```

Comment:
I think the adding a new grid row / collapsible section should updated `gridLayout$` directly (since this action will not be "cancellable") so I think this should still be on `gridLayout$`